### PR TITLE
Multi-targeting host and test projects to support .NET 8

### PR DESCRIPTION
**Pull request by fabiocav**
### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "6.0.419",
+    "version": "8.0.204",
```
Comment (liliankasem) [inline global.json:3]: In dev we are still on `8.0.101` - should we ensure these are aligned?

**Review by liliankasem — APPROVED** [review]
minor comment on sdk version otherwise lgtm

File: global.json (hunk 1 above)
Comment (fabiocav) [reply global.json:3]: I was thinking about that as I was making the change. We can go back now, or keep this in the feature branch as we should have this deployed by the time we need to merge.

Comment (liliankasem) [reply global.json:3]: If 204 will be deployed soon, let's just keep it

**Review by kshyju — APPROVED** [review]

**Pull request merged**